### PR TITLE
Migrate stdlib modules to new table syntax

### PR DESCRIPTION
**Pull request by MadhukaHarith92**
## Purpose
Change io:csv, http:cookie modules to the new table syntax and enable test cases.

Fixes #<Issue Number>
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/22328
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/22329

### CODE REVIEW

File: stdlib/http/src/test/resources/test-src/services/nativeimpl/cookie/cookie-native-function-negative.bal [hunk 1]
```diff
@@ -248,7 +248,6 @@ function testAddPersistentCookieWithoutPersistentStore() returns @tainted http:C
     }
     return cookies;
 }
-
 //TODO Table remove - Fix
 //function testRemovePersistentCookieFromCookieStore_1() returns @tainted http:Cookie[] {
```
Comment (daneshk) [inline stdlib/http/src/test/resources/test-src/services/nativeimpl/cookie/cookie-native-function-negative.bal:252]: Why this test case is commented out?

File: stdlib/http/src/test/java/org/ballerinalang/stdlib/services/nativeimpl/cookie/CookieNativeFunctionNegativeTest.java [hunk 2]
```diff
@@ -157,6 +157,7 @@ public void testAddPersistentCookieWithoutPersistentStore() {
                           "Cookie objects are in the Return Values");
     }
 
+    //TODO Table remove - Fix
```
Comment (daneshk) [inline stdlib/http/src/test/java/org/ballerinalang/stdlib/services/nativeimpl/cookie/CookieNativeFunctionNegativeTest.java:160]: Why TODO? Can we remove it

File: stdlib/io/src/main/ballerina/src/io/readable_csv_channel.bal [hunk 3]
```diff
@@ -94,21 +96,23 @@ public type ReadableCSVChannel object {
...
+# Returns a table, which corresponds to the CSV records.
+# ```ballerina
+# var tblResult1 = readableCSVChannel.getTable(Employee);
+# var tblResult2 = readableCSVChannel.getTable(Employee, ["id", "name"]);
+# ```
+#
+# + structType - The object in which the CSV records should be deserialized
+# + tableKey - The key of the table
+# + return - Table, which represents the CSV records or else an `io:Error`
+    public function getTable(typedesc<record {}> structType, public string[] tableKey = [])
```
Comment (daneshk) [inline stdlib/io/src/main/ballerina/src/io/readable_csv_channel.bal:108]:
```suggestion
    public function getTable(typedesc<record {}> structType, public string[] fieldNames = [])
```

Comment (daneshk) [reply stdlib/io/src/main/ballerina/src/io/readable_csv_channel.bal:108]: Shall we name it as `fieldNames`?

File: stdlib/io/src/main/ballerina/src/io/readable_csv_channel.bal [hunk 4]
```diff
@@ -94,21 +96,23 @@ public type ReadableCSVChannel object {
...
-    //    return getTableExtern(self, structType);
-    //}
+# Returns a table, which corresponds to the CSV records.
+# ```ballerina
+# var tblResult1 = readableCSVChannel.getTable(Employee);
+# var tblResult2 = readableCSVChannel.getTable(Employee, ["id", "name"]);
+# ```
+#
+# + structType - The object in which the CSV records should be deserialized
+# + tableKey - The key of the table
```
Comment (daneshk) [inline stdlib/io/src/main/ballerina/src/io/readable_csv_channel.bal:106]:
```suggestion
# + fieldNames - The names of the fields used as the (composite)key of the table.
```

File: tests/jballerina-unit-test/src/test/resources/test-src/types/string/bstring-table-test.bal [hunk 5]
```diff
@@ -20,14 +20,11 @@ type Employee record {
...
-           [
-               {1, "Mary🤒", 300.5},
-               {2, "John💉", 200.5},
-               {3, "Jim", 330.5}
-           ]
-       };
+   table<Employee> tbEmployee = table [
+               {id: 1, name: "Mary", salary: 300.5},
+               {id: 2, name: "John", salary: 200.5},
+               {id: 3, name: "Jim", salary: 330.5}
```
Comment (daneshk) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/string/bstring-table-test.bal:26]: Why are we changing these test cases?

Comment (MadhukaHarith92) [reply tests/jballerina-unit-test/src/test/resources/test-src/types/string/bstring-table-test.bal:26]: Test cases are the same. Syntax of initializing the table got changed.

File: stdlib/http/src/test/java/org/ballerinalang/stdlib/services/nativeimpl/cookie/CookieNativeFunctionNegativeTest.java (hunk 2 above)
Comment (MadhukaHarith92) [reply stdlib/http/src/test/java/org/ballerinalang/stdlib/services/nativeimpl/cookie/CookieNativeFunctionNegativeTest.java]: Updated in 9a986a0

File: stdlib/http/src/test/resources/test-src/services/nativeimpl/cookie/cookie-native-function-negative.bal (hunk 1 above)
Comment (MadhukaHarith92) [reply stdlib/http/src/test/resources/test-src/services/nativeimpl/cookie/cookie-native-function-negative.bal]: This test case checked the error message when removing a non-existing cookie. However, with the new table, it panics if we try to remove an entry from the table if the key does not exist hence the test case failed. Updated in 9a986a0 to assert the runtime exception.

File: stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/GetTable.java [hunk 6]
```diff
@@ -35,69 +58,107 @@
...
+                return IOUtils.createEoFError();
+            }
+            List<String[]> records = new ArrayList<>();
+            while (delimitedChannel.hasNext()) {
+                records.add(delimitedChannel.read());
+            }
+            return getTable(typedescValue, key, records);
+        } catch (BallerinaIOException | BallerinaException e) {
+            String msg = "failed to process the delimited file: " + e.getMessage();
+            log.error(msg, e);
```
Comment (Bhashinee) [inline stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/GetTable.java:77]: Do we need to log this error here? As we are returning the error, anyway user will have to handle that.

Comment (MadhukaHarith92) [reply stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/GetTable.java]: Updated in 4312c4f

File: tests/jballerina-unit-test/src/test/resources/test-src/types/string/bstring-table-test.bal (hunk 5 above)
Comment (daneshk) [reply tests/jballerina-unit-test/src/test/resources/test-src/types/string/bstring-table-test.bal:26]: "🤒", "💉" are missing in the updated one

Comment (MadhukaHarith92) [reply tests/jballerina-unit-test/src/test/resources/test-src/types/string/bstring-table-test.bal:26]: Didn't know they were intentionally added hence removed. Added back in 70e13a6 .

**Pull request merged**